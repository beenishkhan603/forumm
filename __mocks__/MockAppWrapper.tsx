import { AuthContext } from '@libs/useAuth'
import { MockedProvider, MockedResponse } from '@apollo/client/testing'
import { ProfileInfo } from '@type/Hooks/useAuth/Auth.type'
import ALL_MOCKS from './graphql/mocks'

export const defaultMockProfile: ProfileInfo = {
  company: '448 Studio',
  email: 'test@email.com',
  fullName: 'John Doe',
  university: '448 Studio',
  profileImageUrl: '/test.png',
  userId: '123',
  groups: [],
}

const MockAppWrapper = ({
  children,
  mocks,
  mockProfile = defaultMockProfile,
}: {
  mockProfile?: Partial<ProfileInfo>
  children: any
  mocks?: readonly MockedResponse<Record<string, any>>[] | undefined
}) => {
  const Layout = children?.type?.Layout ?? null

  const GQL_MOCKS: MockedResponse[] = ALL_MOCKS

  if (mocks) mocks.forEach((m) => GQL_MOCKS.push(m))
  return (
    <MockedProvider
      mocks={GQL_MOCKS}
      defaultOptions={{
        query: {
          errorPolicy: 'all',
          fetchPolicy: 'cache-first',
        },
        watchQuery: {
          errorPolicy: 'all',
          fetchPolicy: 'cache-and-network',
        },
        mutate: {
          errorPolicy: 'all',
        },
      }}
    >
      <AuthContext.Provider
        value={{
          profile: mockProfile
            ? { ...defaultMockProfile, ...mockProfile }
            : defaultMockProfile,
          setProfile: () => {},
        }}
      >
        <div id="modal-root" />
        {Layout ? <Layout>{children}</Layout> : children}
      </AuthContext.Provider>
    </MockedProvider>
  )
}

export default MockAppWrapper
