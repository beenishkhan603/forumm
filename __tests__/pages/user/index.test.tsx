import { render } from '@testing-library/react'
import UserDetailsPage from 'pages/settings'
import MockAppWrapper from '__mocks__/MockAppWrapper'

it('renders user index page unchanged', () => {
  const { container } = render(
    <MockAppWrapper>
      <UserDetailsPage />
    </MockAppWrapper>
  )
  expect(container).toMatchSnapshot()
})
