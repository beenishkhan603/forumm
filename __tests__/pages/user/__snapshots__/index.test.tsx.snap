// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders user index page unchanged 1`] = `
<div>
  <div
    id="modal-root"
  />
  <div
    class="flex justify-center"
    style="color: rgb(24, 26, 32);"
  >
    <div
      class="flex flex-col items-center mt-14 text-white max-w-md w-full "
      style="color: rgb(24, 26, 32);"
    >
      <div
        class="text-3xl"
        style="color: rgb(24, 26, 32);"
      >
        User Profile 
      </div>
      <img
        alt="Profile Image"
        class="w-40 h-40 bg-contain bg-center rounded-full object-cover mt-8 border-2"
        data-nimg="1"
        decoding="async"
        height="160"
        loading="lazy"
        src="/_next/image?url=%2Ftest.png&w=384&q=75"
        srcset="/_next/image?url=%2Ftest.png&w=256&q=75 1x, /_next/image?url=%2Ftest.png&w=384&q=75 2x"
        style="color: transparent; border-color: #3763e9;"
        width="160"
      />
      <div
        class="text-white text-xl mt-8"
        style="color: rgb(24, 26, 32);"
      >
        John Doe
      </div>
      <div
        class="text-white text-sm"
        style="color: rgb(24, 26, 32);"
      >
        test@email.com
      </div>
      <div
        class="text-white mt-8"
        style="color: rgb(24, 26, 32);"
      >
        Member
         Account
      </div>
      <div
        class="text-white text-sm"
        style="color: rgb(24, 26, 32);"
      >
        448 Studio
      </div>
       
      <div
        class="flex space-x-4 my-8 w-full"
        style="color: rgb(24, 26, 32);"
      >
        <button
          class="
      bg-forumm-blue-2 text-white relative rounded-full  px-6 py-4 text-base h-8 flex items-center hover:animate-heartbeat justify-center flex-1"
          style="background-color: rgb(55, 99, 233);"
          type="button"
        >
          <span
            class="relative z-10"
            style="color: white;"
          >
            Enable Anonymous Mode
          </span>
        </button>
      </div>
      <div
        class="flex space-x-4 my-8 w-full"
        style="color: rgb(24, 26, 32);"
      >
        <button
          class="
      bg-forumm-blue-2 text-white relative rounded-full  px-6 py-4 text-base h-8 flex items-center hover:animate-heartbeat justify-center flex-1"
          style="background-color: rgb(55, 99, 233);"
          type="button"
        >
          <span
            class="relative z-10"
            style="color: white;"
          >
            Edit Profile
          </span>
        </button>
        <button
          class="
      bg-forumm-blue-2 text-white relative rounded-full  px-6 py-4 text-base h-8 flex items-center hover:animate-heartbeat justify-center flex-1"
          style="background-color: rgb(55, 99, 233);"
          type="button"
        >
          <span
            class="relative z-10"
            style="color: white;"
          >
            Change Password
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
